Formula/h/hdf5.rb

```diff
@@ -124,3 +121,37 @@ def install
     assert_match %r{I/O filters.*DECODE}, config
   end
 end
+
+__END__
+diff --git a/CMakeFilters.cmake b/CMakeFilters.cmake
+index 52d65e5..b43aa3b 100644
+--- a/CMakeFilters.cmake
++++ b/CMakeFilters.cmake
+@@ -108,7 +108,7 @@ if (HDF5_ENABLE_Z_LIB_SUPPORT)
+         # on the target. The target returned is: ZLIB::ZLIB
+         get_filename_component (libname ${ZLIB_LIBRARIES} NAME_WLE)
+         string (REGEX REPLACE "^lib" "" libname ${libname})
+-        set_target_properties (ZLIB::ZLIB PROPERTIES OUTPUT_NAME zlib-static)
++        set_target_properties (ZLIB::ZLIB PROPERTIES OUTPUT_NAME z)
```

Comment:
Upstream fix would probably be
```suggestion
+        set_target_properties (ZLIB::ZLIB PROPERTIES OUTPUT_NAME ${libname})
```

Reply:
Upstreamed this one. Other patch is taking a while upstream.